docs/modelserving/servingruntimes.md

```diff
@@ -0,0 +1,179 @@
+| `tolerations`                      | Allow pods to be scheduled onto nodes [with matching taints](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration)                                                 |
+
+ModelMesh leverages additional fields not listed here.  More information [here](https://github.com/kserve/modelmesh-serving/blob/main/docs/runtimes/custom_runtimes.md#spec-attributes).
+
+## Using ServingRuntimes
+
+ServingRuntimes can be be used both explicitly and implicitly.
+
+### Explicit: specify a runtime
+
+When users define predictors in their _InferenceServices_, they can explicitly specify the name of a _ClusterServingRuntime_ or _ServingRuntime_. For example:
+
+```yaml
+apiVersion: serving.kserve.io/v1beta1
+kind: InferenceService
+metadata:
+  name: example-sklearn-isvc
+spec:
+  predictor:
+    model:
+      modelFormat:
+        name: sklearn
+      storageUri: s3://bucket/sklearn/mnist.joblib
+      runtime: kserve-mlserver
+```
+
+Here, the runtime specified is `kserve-mlserver`, so the KServe controller will first search the namespace for a _ServingRuntime_ with that name. If
+none exist, the controller will then search the list of _ClusterServingRuntimes_. If one is found, the the controller will first
+verify that the `modelFormat` provided in the predictor is in the list of `supportedModelFormats`. If it is, then the container and pod information provided
+by the runtime will be used for model deployment.
+
+### Implicit: Automatic selection
+
+In each entry of the `supportedModelFormats` list, `autoSelect: true` can optionally be specified to indicate that that the given `ServingRuntime` can be
+considered for automatic selection for predictors with the corresponding model format if no runtime is explicitly specified.
+For example, the `kserve-sklearnserver` _ClusterServingRuntime_ supports SKLearn version 1 and has `autoSelect` enabled:
+
+```yaml
+apiVersion: serving.kserve.io/v1alpha1
+kind: ClusterServingRuntime
+metadata:
+  name: kserve-sklearnserver
+spec:
+  supportedModelFormats:
+    - name: sklearn
+      version: "1"
+      autoSelect: true
+...
+```
+
+When the following InferenceService is deployed with no runtime specified, the controller will look for a runtime that supports `sklearn`:
+
+```yaml
+apiVersion: serving.kserve.io/v1beta1
+kind: InferenceService
+metadata:
+  name: example-sklearn-isvc
+spec:
+  predictor:
+    model:
+      modelFormat:
+        name: sklearn
+      storageUri: s3://bucket/sklearn/mnist.joblib
+```
+
+Since `kserve-sklearnserver` has an entry in its `supportedModelFormats` list with `sklearn` and `autoSelect: true`, this _ClusterServingRuntime_
+will be used for model deployment.
+If multiple runtimes have `sklearn` as an auto-selectable supported format, then there is no guarantee _which_ runtime will be selected.
```

Comment:
Can we add this to a warning section with `!!! warning`?

Reply:
Sure, added